device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -354,15 +354,17 @@ public Integer sendMessage(Message message)
             byte[] tag = String.valueOf(this. nextTag++).getBytes();
             Delivery dlv = sender.delivery(tag);
 
```

Comment:
Can we add logging on line 345 just before breaking that encoding is complete ? I suspect the issue is somewhere in this block.